### DISCUSSION TITLE
bug: animate() doesn't work if called with a delay after the animate component is created

### DISCUSSION THREAD

**Discussion by themerk** [kaplayjs/kaplay#556] [Q&A] [answered]
## **Describe the bug**

Animations (specifically with `{ loops: 1 }`) work as expected if started at the same time as the game object is created.

But if animations are applied later (either with `wait()`, or on a key press etc) the object just jumps to the final keyframe, without any animation.

The problem appears to be the initial timestamp being set when the `animate` component is created, rather than when the animation is started:
https://github.com/kaplayjs/kaplay/blob/b89b97c12a2e5864d46f31818e85edcc81b39175/src/components/misc/animate.ts#L509-L511

This would probably work ok if `.use(animate())` is called just before applying the animation, but that's a bit unintuitive and a real pain when an animation is a one-shot, non-looping sequence, like you might have on a button press or collision. Or if you try to animate multiple properties with different, overlapping start times.

## **Version**

Replicated in both 3001.0.2 and 4000.0.0-alpha15

## **To Reproduce**

[Example on KAPLAYGROUND](https://play.kaplayjs.com/?code=eJytUbFuwjAU3PMVJyZHikgCykJFh6osVadKVQfE8HAcYnDiyHaaIsS%2F1zGl7VaGSh7e6b13d76Xplh9UNMpAV2BWtmQk7q1aLXDoM1BtjsMtWhhHRknSlDlhIGrBXbU%2BK294A7SghtBvh9FB%2BoUHdkpAgZZunqBIssSj2ohd7X7guf4Loq4V3IwnnUJKku29lNcGq4Em2fxuMO10oa9rB4D6rRleZElmBWXtuXkZ0N58e5BtPmh3qpe%2FMH98Py6%2Biaf30aepnjz2VjoA7aCU2%2FFr0S2IZKBfLtVR%2Bx7b%2BSajv%2Fr9Eo2CQKTBOt3wWcsjxOEYhZvEpxQ9ibcYoFsWiRQWnd2gRznOBh4Gmn3fdNZOB3UK9mSQmW8iWgg6ViegMVY3mO8xRjFP0j79wkxXqhS)

```js
kaplay({
  width: 500,
  height: 500,
});

const red = add([
  circle(30),
  color(RED),
  pos(150, 250),
  scale(),
  animate()
]);

const blue = add([
  circle(30),
  color(BLUE),
  pos(350, 250),
  scale(),
  animate()
]);

// Works ok because the game object was only just created
red.animate("scale", [vec2(1), vec2(2)], { duration: 0.5, loops: 1 })

// Just jumps to the final frame
wait(1, () => {
  blue.animate("scale", [vec2(1), vec2(2)], { duration: 0.5, loops: 1 })
})
```

## **Expected behavior**

If a game object has the `animate()` component added during its creation, we should be free to call `.animate` at any time in the future and have the animation play as expected.

**Comment by mflerackers** [comment, marked as answer]
There's a seek method for that reason, to seek to time 0.

**Comment by themerk** [comment]
Ah I see. I don't think that's mentioned in the docs.

I guess it just comes down to unintuitive behaviour for someone new to the library. It makes sense to have a seek function for advanced manipulation of an animation. But I just expected the animate function to... animate. Since my first use of it was on a button press, it wasn't at all obvious why I was getting that behaviour.

Should any call to animate() just automatically seek to time 0?

**Comment by mflerackers** [comment]
I'll rather leave the control to the user. Animate is originally made to define an animation, then play it.
